flink-table/flink-table-api-java-bridge/src/main/java/org/apache/flink/table/sinks/CsvTableSinkFactoryBase.java

```diff
@@ -75,6 +75,8 @@
 		properties.add(SCHEMA + ".#." + DescriptorProperties.TABLE_SCHEMA_TYPE);
 		properties.add(SCHEMA + ".#." + DescriptorProperties.TABLE_SCHEMA_DATA_TYPE);
 		properties.add(SCHEMA + ".#." + DescriptorProperties.TABLE_SCHEMA_NAME);
+		// schema watermark
+		properties.add(SCHEMA + "." + DescriptorProperties.WATERMARK + ".*");
```

Comment:
Why need watermark?

Reply:
The factory can't be found if a watermark is defined in DDL. 
I will update the IT case to use watermark.

Reply:
So every connector(SourceTableFactory and SinkTableFactory) must has this?
Kafka not work too?

Reply:
Yes. I will fix it in following PR with some computed column problems.

Reply:
Can you create a JIRA to track it too? Should in 1.10?

Reply:
issue created FLINK-15191.

Reply:
side comment: at some point we need to refactor this such that not every connector much list all properties. This is also why we introduced `TableSourceFactory#createTableSource(org.apache.flink.table.catalog.ObjectPath, org.apache.flink.table.catalog.CatalogTable)`
The outer layer should deal with the schema properties and the connector can work with `CatalogTable.getTableSchema` if necessary.

Reply:
Exactly, agree.